utils/service_discovery/zookeeper_config_store.py

```diff
@@ -0,0 +1,99 @@
+# (C) Datadog, Inc. 2016
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# std
+import logging
+
+from kazoo.client import KazooClient, NoNodeError
+from utils.service_discovery.abstract_config_store import AbstractConfigStore, KeyNotFound
+
+DEFAULT_ZK_HOST = '127.0.0.1'
+DEFAULT_ZK_PORT = 2181
+DEFAULT_TIMEOUT = 5
+log = logging.getLogger(__name__)
+
+class ZookeeperStore(AbstractConfigStore):
+    """Implementation of a config store client for Zookeeper"""
+
+    def _extract_settings(self, config):
+        """Extract settings from a config object"""
+        settings = {
+            'host': config.get('sd_backend_host', DEFAULT_ZK_HOST),
+            'port': int(config.get('sd_backend_port', DEFAULT_ZK_PORT)),
+        }
+        return settings
+
+    def get_client(self, reset=False):
+        if self.client is None or reset is True:
+            self.client = KazooClient(
+                hosts=self.settings.get('host') + ":" + str(self.settings.get('port')),
+                read_only=True,
+            )
+            self.client.start()
+        return self.client
+
+    def client_read(self, path, **kwargs):
+        """Retrieve a value from a Zookeeper key."""
+        try:
+            if kwargs.get('watch', False):
+                return self.recursive_mtime(path)
+            elif kwargs.get('all', False):
+                # we use it in _populate_identifier_to_checks
+                results = []
+                self.recursive_list(path, results)
+                return results
+            else:
+                res, stats = self.client.get(path)
+                return res.decode("utf-8")
+        except NoNodeError:
+            raise KeyNotFound("The key %s was not found in Zookeeper" % path)
+
+    def recursive_list(self, path, results):
+        """Recursively walks the children from the given path and build a list of key/value tuples"""
+        try:
+            data, stat = self.client.get(path)
+
+            if data:
+                node_as_string = data.decode("utf-8")
+                if node_as_string[0] == '[':
```

Comment:
@manolama would an empty string work for you here instead of `'['`? This would allow us to do `if not node_as_string` (cleaner) and would avoid swallowing formatting errors (people forgetting that they need to use a list instead of a single dict, etc.). If that works for you I can add this change and merge your PR.